### PR TITLE
Release/2.5.0

### PR DESCRIPTION
**Pull request by bdice**
Release 2.5.0.

### CODE REVIEW

**Comment by bdice** [comment]
@vyasr @joaander I'm getting an error for Mac PyPI wheels. If you have time, I'd appreciate help debugging this. I know Vyas has spent considerable time on these deployment scripts. I wonder if cibuildwheel would be easier or harder to use...

```
Traceback (most recent call last):
  File "/Users/distiller/.pyenv/versions/3.9.1/bin/delocate-wheel", line 8, in <module>
    sys.exit(main())
  File "/Users/distiller/.pyenv/versions/3.9.1/lib/python3.9/site-packages/delocate/cmd/delocate_wheel.py", line 69, in main
    copied = delocate_wheel(wheel, out_wheel, lib_filt_func=lib_filt_func,
  File "/Users/distiller/.pyenv/versions/3.9.1/lib/python3.9/site-packages/delocate/delocating.py", line 377, in delocate_wheel
    copied_libs = delocate_path(package_path, lib_path,
  File "/Users/distiller/.pyenv/versions/3.9.1/lib/python3.9/site-packages/delocate/delocating.py", line 289, in delocate_path
    copied = delocate_tree_libs(lib_dict, lib_path, tree_path)
  File "/Users/distiller/.pyenv/versions/3.9.1/lib/python3.9/site-packages/delocate/delocating.py", line 78, in delocate_tree_libs
    raise DelocationError('Already planning to copy library with '
delocate.delocating.DelocationError: Already planning to copy library with same basename as: libtbb.12.2.dylib
```

File: .circleci/deploy-macos.sh [hunk 1]
```diff
@@ -33,18 +33,25 @@ export MACOSX_DEPLOYMENT_TARGET=10.12
 # tries to delete them again and causes errors. To avoid this issue, we update
 # manually here for the current main version of Python (required by pyenv).
 brew upgrade python@3.8
```
Comment (bdice) [inline .circleci/deploy-macos.sh:35]: Try this if Mac builds fail.
```suggestion
brew upgrade python@3.9
```

File: freud/CMakeLists.txt [hunk 2]
```diff
@@ -86,8 +86,6 @@ foreach(cython_module ${cython_modules_with_cpp} ${cython_modules_without_cpp})
   else()
     set_target_properties(${cython_module} PROPERTIES INSTALL_RPATH "\$ORIGIN")
   endif()
```
Comment (bdice) [inline freud/CMakeLists.txt:104]: Revert this if Mac builds fail.
```suggestion
  endif()
  set_target_properties(${cython_module} PROPERTIES INSTALL_RPATH_USE_LINK_PATH
                                                    True)
```

**Comment by vyasr** [comment]
Last error looks the usual numpy ABI incompatibility. Possibly triggered by the installation of the freud wheel uninstalling numpy 1.14.6 and downgrading to 1.14.0, especially since the error is in gsd and not freud.

**Comment by vyasr** [comment]
I think the current delocate warnings are what the `INSTALL_RPATH_USE_LINK_PATH` might be necessary to fix on Mac, but I can't remember for sure. If we can get the wheels past the current numpy issue and then the freud import fails with undefined symbols when we would know.

**Comment by vyasr** [comment]
I believe this is ready to go once we've confirmed that the conda fix works, right? Then we can look into cibuildwheel for future releases (cf. glotzerlab/gsd#89).

**Comment by joaander** [comment]
> I believe this is ready to go once we've confirmed that the conda fix works, right? Then we can look into cibuildwheel for future releases (cf. [glotzerlab/gsd#89](https://github.com/glotzerlab/gsd/pull/89)).

We will need to enable the tests that were commented out to reduce latency.

**Comment by bdice** [comment]
@klywang confirmed that the current branch works for her. I'm going to finalize this PR and make the release.

**Comment by bdice** [comment]
Currently the manylinux Docker container is failing with this error.
```
Error pulling image quay.io/pypa/manylinux2010_x86_64: Error response from daemon: received unexpected HTTP status: 502 Bad Gateway... retrying
  image cache not found on this host, downloading quay.io/pypa/manylinux2010_x86_64
```
The [quay.io status](https://status.quay.io/history) indicates there is a server issue.
image

We'll need to wait until this is resolved before the builds will pass.

**Comment by joaander** [comment]
Now we get: `fatal: Remote branch issue136 not found in upstream origin`

File: .circleci/deploy-linux.sh [hunk 3]
```diff
@@ -44,18 +50,17 @@ for PYBIN in $PYBINS; do
...
-  rm -f PKG-INFO
-  "${PYBIN}/python" -m pip install . --no-deps --ignore-installed -v --progress-bar=off -q
+  "${PYBIN}/python" -m pip install cython scikit-build cmake oldest-supported-numpy --ignore-installed -q --progress-bar=off
   "${PYBIN}/pip" wheel ~/ci/freud/ -w ~/wheelhouse/ --no-deps --no-build-isolation --no-use-pep517
 done
 
-# Update RPath for wheels
+# Install patched auditwheel (fixes RPATHs for libfreud/libtbb, issue #136).
+cd ~/
+git clone https://github.com/mayeut/auditwheel.git -b issue136 auditwheel
```
Comment (bdice) [inline .circleci/deploy-linux.sh]: @joaander This branch was merged and deleted. We can install from the upstream's `master` branch now.
```suggestion
git clone https://github.com/pypa/auditwheel.git -b master auditwheel
```

**Pull request merged**